include/mbedtls/md.h

```diff
@@ -49,6 +49,12 @@ typedef enum {
     MBEDTLS_MD_RIPEMD160,
 } mbedtls_md_type_t;
 
+typedef enum {
+    MBEDTLS_SIG_RSA=0,
+    MBEDTLS_SIG_ECDSA,
+    MBEDTLS_SIG_NONE,
```

Comment:
The type isn't public in the sense that nobody outside us is meant to use these constants. The values aren't part of the ABI. It would be best if this type could be restricted to a single source file.

Instead of having `NONE` at the end, maybe it should be -1? And there should be a different constant `MBEDTLS_SIG_ARRAY_SIZE` for the array size (a good idea for clarity anyway).